### PR TITLE
Applied the feedback from community to our extension

### PR DESCRIPTION
**Pull request by jerrytfleung**
Applied the [PR](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30788) feedback from community to our extension.

1. It fixed the shutdown problem.
2. Changed noop to getting soft disabled settings from collector
3. Reverted validate method to config

### CODE REVIEW

File: collector/extension/solarwindsapmsettingsextension/config.go [hunk 1]
```diff
@@ -1,9 +1,67 @@
...
+	"go.opentelemetry.io/collector/component"
+)
 
 type Config struct {
 	Endpoint string        `mapstructure:"endpoint"`
 	Key      string        `mapstructure:"key"`
 	Interval time.Duration `mapstructure:"interval"`
 }
+
+const (
```
Comment (swi-jared) [inline collector/extension/solarwindsapmsettingsextension/config.go:18]: Do these `const`s need to be public? If not, I'd make them private by lower-casing the first letter.

File: collector/extension/solarwindsapmsettingsextension/config.go [hunk 2]
```diff
@@ -1,9 +1,67 @@
...
+		cfg.Interval = MinimumInterval
+	}
+	if cfg.Interval.Seconds() > MaximumInterval.Seconds() {
+		cfg.Interval = MaximumInterval
+	}
+	return nil
+}
+
+func resolveServiceNameBestEffort() string {
+	if otelServiceName, otelServiceNameDefined := os.LookupEnv("OTEL_SERVICE_NAME"); otelServiceNameDefined && len(otelServiceName) > 0 {
```
Comment (swi-jared) [inline collector/extension/solarwindsapmsettingsextension/config.go]: Idiomatically, it's OK to use and re-use names. For this and the similar version below I'd stick with `if name, ok :=`

**Review by swi-jared — APPROVED** [review]
Minor nits, looks good otherwise

File: collector/extension/solarwindsapmsettingsextension/config.go (hunk 1 above)
Comment (jerrytfleung) [reply collector/extension/solarwindsapmsettingsextension/config.go:18]: Yes, all examples from otel collector & otel collector contrib repo are in capital letter

File: collector/extension/solarwindsapmsettingsextension/config.go (hunk 2 above)
Comment (jerrytfleung) [reply collector/extension/solarwindsapmsettingsextension/config.go]: Thanks! Fixed!

**Pull request merged**